test/unit/helpers/QtiPackageExporterTest.php

```diff
@@ -0,0 +1,144 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA ;
+ */
+declare(strict_types=1);
+
+namespace oat\taoQtiTest\test\unit\helpers;
+
+use common_Exception;
+use common_report_Report;
+use oat\oatbox\filesystem\Directory;
+use oat\oatbox\filesystem\File;
+use oat\oatbox\filesystem\FileSystemService;
+use oat\tao\helpers\FileHelperService;
+use taoQtiTest_models_classes_export_TestExport22;
+use oat\generis\test\MockObject;
+use oat\generis\test\TestCase;
+use oat\taoQtiTest\helpers\QtiPackageExporter;
+
+class QtiPackageExporterTest extends TestCase
+{
+    /** @var QtiPackageExporter */
+    private $subject;
+
+    /** @var taoQtiTest_models_classes_export_TestExport22|MockObject */
+    private $exporterMock;
+
+    /** @var FileSystemService|MockObject */
+    private $fileSystemServiceMock;
+
+    /** @var FileHelperService|MockObject */
+    private $fileHelperServiceMock;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+        $this->exporterMock = $this->createMock(taoQtiTest_models_classes_export_TestExport22::class);
+        $this->fileSystemServiceMock = $this->createMock(FileSystemService::class);
+        $this->fileHelperServiceMock = $this->createMock(FileHelperService::class);
+        $this->fileHelperServiceMock->method('createTempDir')->willReturn('FAKE_TMP_DIR');
+
+        $this->subject = new QtiPackageExporter(
+            $this->exporterMock,
+            $this->fileSystemServiceMock,
+            $this->fileHelperServiceMock
+        );
+    }
+
+    public function testExportDeliveryQtiPackage_ThrowsExceptionWhenExportFails(): void
+    {
+        $testUri = 'FAKE_TEST_URI';
+
+        $expectedReport = common_report_Report::createFailure('FAKE ERROR MESSAGE');
+        $this->exporterMock->method('export')
+            ->willReturn($expectedReport);
+
+        $this->expectException(common_Exception::class);
```

Comment:
I would suggest to assert on the exception message as well

Reply:
I can add it, but I don't like assertions on exception messages. When message changes test will file, but it will not affect business logic if exception type is the same.